test/parallel/test-worker-connection-errors.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+const common = require('../common');
+const {
+  connect,
+  isMainThread,
+  setConnectionsListener,
+  workerData,
+  Worker,
+} = require('node:worker_threads');
+const { throws } = require('node:assert');
+
+if (isMainThread) {
+  throws(common.mustCall(function() {
+    connect(0);
+  }), {
+    name: 'Error',
+    code: 'ERR_WORKER_SAME_THREAD',
+    message: 'Cannot connect to the same thread'
```

Comment:
We almost always never check for error messages since they're not bound by the semantic version spec.

```suggestion
```

Reply:
Understood, removed.